### PR TITLE
[ConstraintSystem] NFC: Define {activate, deactivate}Constraint to make…

### PR DESCRIPTION
**Pull request by xedin**
… that logic reusable

### CODE REVIEW

**Comment by xedin** [comment]
@rudkx Had this refactoring as part of the missing member work which is ongoing, decided to push separately.

**Comment by xedin** [comment]
@swift-ci please smoke test

File: lib/Sema/CSSolver.cpp [hunk 1]
```diff
@@ -269,36 +268,21 @@ bool ConstraintSystem::simplify(bool ContinueAfterFailures) {
...
       if (solverState)
         ++solverState->NumUnsimplifiedConstraints;
-
-      InactiveConstraints.push_back(constraint);
       break;
     }
 
-    // This constraint is not active. We delay this operation until
-    // after simplification to avoid re-insertion.
-    constraint->setActive(false);
```
Comment (rudkx) [inline lib/Sema/CSSolver.cpp]: It's not clear to me what the comment is referring to, but it seems like this was being done later for a reason.

Are you sure we don't still need to delay this to this point?

Comment (xedin) [reply lib/Sema/CSSolver.cpp]: I think this is outdated, it's left here from SVN days when simplication was completely different, we don't check isActive() anymore besides two places in the solver, this one being one of them and another one is an assert.

**Review by rudkx — APPROVED** [review]
LGTM!

**Comment by xedin** [comment]
SwiftPM test failure on linux.

**Comment by xedin** [comment]
@swift-ci please smoke test Linux platform

**Comment by xedin** [comment]
@swift-ci please smoke test Linux platform

**Comment by xedin** [comment]
@swift-ci please test source compatibility

File: lib/Sema/ConstraintSystem.h [hunk 2]
```diff
@@ -1996,6 +1996,28 @@ class ConstraintSystem {
...
+  /// Transfer given constraint from to active list
+  /// for solver to attempt its simplification.
+  void activateConstraint(Constraint *constraint) {
+    assert(!constraint->isActive() && "Constraint is already active");
+    ActiveConstraints.splice(ActiveConstraints.end(), InactiveConstraints,
+                             constraint);
+    constraint->setActive(true);
+  }
+
+  void deactiveConstraint(Constraint *constraint) {
```
[the remark this answers is not in the document]
Comment (xedin) [reply lib/Sema/ConstraintSystem.h]: That's right, thank you!

**Pull request merged**